docs/learn/expert-guides/deploy-multichain-contracts.md

```diff
@@ -0,0 +1,581 @@
+      "outputs": [
+        {
+          "internalType": "uint256",
+          "name": "",
+          "type": "uint256"
+        }
+      ],
+      "stateMutability": "view",
+      "type": "function"
+    },
+    {
+      "inputs": [
+        {
+          "internalType": "uint256",
+          "name": "_number",
+          "type": "uint256"
+        }
+      ],
+      "name": "setNumber",
+      "outputs": [],
+      "stateMutability": "nonpayable",
+      "type": "function"
+    }
+  ],
+  "bytecode": "0x608060405234801561001057600080fd5b5060405161010f38038061010f83398101604081905261002f91610037565b600055610050565b60006020828403121561004957600080fd5b5051919050565b60b18061005e6000396000f3fe6080604052348015600f57600080fd5b506004361060325760003560e01c80633fb5c1cb1460375780638381f58a146049575b600080fd5b604760423660046063565b600055565b005b605160005481565b60405190815260200160405180910390f35b600060208284031215607457600080fd5b503591905056fea2646970667358221220958426080fb00e4b3e137866f9d1884559c2a31f2d3ee15cd4f2c8aee4a92e6764736f6c63430008110033"
+}
+```
+
+</details>
+
+This file is essential for the upcoming steps and should be moved to the same directory as `main.js`.
+
+Once you have the ABI and the bytecode of the contract to deploy, you can start writing your script. We will use the `main.js` file in a new repo.
+
+## Setup
+
+Install `ethers` and `@lukso/lsp-smart-contracts`:
+
+```bash
+npm i ethers @lukso/lsp-smart-contracts
+```
+
+## Step 1: Check the Existence of the Factory
+
+First, establish a connection to the first blockchain network you intend to deploy your contracts on (we will use [LUKSO Testnet](../../networks/testnet/parameters.md)):
+
+```js title="main.js"
+import { ethers } from 'ethers';
+
+const RPC_URL = 'https://rpc.testnet.lukso.network';
+const provider = new ethers.JsonRpcProvider(RPC_URL);
+
+const checkDeployedCode = async (address: any) => {
+  const code = await provider.getCode(address);
+  return code !== '0x';
+};
+
+// Fixed addresses
+
+// For more information check: https://github.com/Arachnid/deterministic-deployment-proxy
+const NICK_FACTORY_ADDRESS = '0x4e59b44847b379578588920ca78fbf26c0b4956c';
+
+// For more information check: https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-16-UniversalFactory.md
+const LSP16_FACTORY_ADDRESS = '0x1600016e23e25D20CA8759338BfB8A8d11563C4e';
+
+const isNickFactoryDeployed = await checkDeployedCode(NICK_FACTORY_ADDRESS);
+const isLSP16FactoryDeployed = await checkDeployedCode(LSP16_FACTORY_ADDRESS);
+
+console.log('Nick Factory exists: ', isNickFactoryDeployed);
+console.log('LSP16UniversalFactory exists: ', isLSP16FactoryDeployed);
+```
+
+### Deployment of Nick Factory
+
+If `Nick Factory` doesn't exist, we'll need to deploy it:
+
+```js title="main.js"
+const signer = new ethers.Wallet('<private-key>', provider);
+
+if (!isNickFactoryDeployed) {
+  const fundingTx = await signer.sendTransaction({
+    to: '0x3fab184622dc19b6109349b94811493bf2a45362',
```

Comment:
What does this address represent?

How can I be sure that if i am on a totally new network, this address exists?

Reply:
Its a standardized address, but i will add acomment